vulnerabilities/package_managers_2.py

```diff
@@ -0,0 +1,377 @@
+        versions = set()
+        response = get_response(url=url)
+        if not response:
+            self.cache[pkg] = versions
+            return
+        for version in response.get("versions", []):
+            release_date = response.get("time", {}).get(version)
+            if release_date:
+                release_date = dateparser.parse(release_date)
+                versions.add(LegacyVersion(value=version, release_date=release_date))
+            else:
+                versions.add(LegacyVersion(value=version, release_date=None))
+
+        self.cache[pkg] = versions
+
+
+class MavenVersionAPI(VersionAPI):
+
+    package_type = "maven"
+
+    def fetch(self, pkg) -> None:
+        artifact_comps = pkg.split(":")
+        endpoint = self.artifact_url(artifact_comps)
+
+        resp = get_response(url=endpoint, type="read")
+
+        if not resp:
+            self.cache[pkg] = set()
+            return
+
+        xml_resp = ET.ElementTree(ET.fromstring(resp.decode("utf-8")))
+        self.cache[pkg] = self.extract_versions(xml_resp)
+
+    @staticmethod
+    def artifact_url(artifact_comps: List[str]) -> str:
+        base_url = "https://repo1.maven.org/maven2/{}"
+        try:
+            group_id, artifact_id = artifact_comps
+        except ValueError:
+            if len(artifact_comps) == 1:
+                group_id = artifact_comps[0]
+                artifact_id = artifact_comps[0].split(".")[-1]
+
+            elif len(artifact_comps) == 3:
+                group_id, artifact_id = list(dict.fromkeys(artifact_comps))
+
+            else:
+                raise
+
+        group_url = group_id.replace(".", "/")
+        suffix = group_url + "/" + artifact_id + "/" + "maven-metadata.xml"
+        endpoint = base_url.format(suffix)
+
+        return endpoint
+
+    @staticmethod
+    def extract_versions(xml_response: ET.ElementTree) -> Set[LegacyVersion]:
+        all_versions = set()
+        for child in xml_response.getroot().iter():
+            if child.tag == "version" and child.text:
+                all_versions.add(LegacyVersion(child.text))
+
+        return all_versions
+
+
+class NugetVersionAPI(VersionAPI):
+
+    package_type = "nuget"
+
+    def fetch(self, pkg) -> None:
+        endpoint = self.nuget_url(pkg)
+        resp = get_response(url=endpoint)
+        if not resp:
+            self.cache[pkg] = set()
+            return
+        self.cache[pkg] = self.extract_versions(resp)
+
+    @staticmethod
+    def nuget_url(pkg_name: str) -> str:
+        pkg_name = pkg_name.lower().strip()
+        base_url = "https://api.nuget.org/v3/registration5-semver1/{}/index.json"
+        return base_url.format(pkg_name)
+
+    @staticmethod
+    def extract_versions(resp: dict) -> Set[LegacyVersion]:
+        all_versions = set()
+        try:
+            for entry_group in resp["items"]:
+                for entry in entry_group["items"]:
+                    all_versions.add(
+                        LegacyVersion(
```

Comment:
I reckon this is old code carried over, but since we are touching it, what about extracting the two duplicated calls below in a variable?
```suggestion
                        catalog_entry = entry["catalogEntry"]
                        LegacyVersion(
```